conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,537 @@
+                    data = load(dst_name)
+                    os.unlink(dst_name)
+                    os.symlink(data, dst_name)
+                else:
+                    f.extract(zi, path=d)
+                    dt = time.mktime(zi.date_time + (0, 0, -1))
+                    os.utime(os.path.join(d, zi.filename), (dt, dt))
+                    if os.name == 'posix':
+                        permissions = zi.external_attr >> 16
+                        os.chmod(os.path.join(d, zi.filename), permissions)
+
+
+class TarCompressor(Compressor):
+    read_mode = "r"
+    write_mode = "w"
+    ext = ".tar"
+
+    def compress(self, d):
+        with tarfile.open(name='test' + self.ext, mode=self.write_mode, format=tarfile.GNU_FORMAT) as f:
+            inodes = dict()
+            for root, dirnames, filenames in os.walk(d):
+                for filename in filenames:
+                    filename = os.path.join(root, filename)
+                    archname = os.path.relpath(filename, d)
+
+                    ti = tarfile.TarInfo(archname)
+                    if os.path.islink(filename):
+                        ti.type = tarfile.SYMTYPE
+                        ti.linkname = os.readlink(filename)
+                        ti.size = 0
+                        f.addfile(tarinfo=ti)
+                    else:
+                        st = os.stat(filename)
+                        if st.st_ino in inodes:
+                            ti.type = tarfile.LNKTYPE
+                            ti.linkname = inodes[st.st_ino]
+                            ti.size = 0
+                            ti.mode = st.st_mode
+                            ti.mtime = st.st_mtime
+                            f.addfile(tarinfo=ti)
+                        else:
+                            inodes[st.st_ino] = archname
+                            ti.size = st.st_size
+                            ti.mode = st.st_mode
+                            ti.mtime = st.st_mtime
+                            with open(filename, 'rb') as file_handler:
+                                f.addfile(tarinfo=ti, fileobj=file_handler)
+                for dirname in dirnames:
+                    filename = os.path.join(root, dirname)
+                    archname = os.path.relpath(filename, d)
+                    ti = tarfile.TarInfo(archname)
+                    if os.path.islink(filename):
+                        ti.type = tarfile.SYMTYPE
+                        ti.linkname = os.readlink(filename)
+                        ti.size = 0
+                        f.addfile(tarinfo=ti)
+                    else:
+                        st = os.stat(filename)
+                        ti.type = tarfile.DIRTYPE
+                        ti.size = 0
+                        ti.mode = st.st_mode
+                        f.addfile(tarinfo=ti)
+
+    def decompress(self, d):
+        with tarfile.open(name='test' + self.ext, mode=self.read_mode) as f:
+            f.extractall(path=d)
```

Comment:
it goes back to #91 (5 years ago).
there are several tricky cases described in https://www.gnu.org/software/tar/manual/html_node/absolute.html:
```
When tar extracts archive members from an archive, it strips any leading slashes (`/') from the member name. This causes absolute member names in the archive to be treated as relative file names. This allows you to have such members extracted wherever you want, instead of being restricted to extracting the member in the exact directory named in the archive. For example, if the archive member has the name `/etc/passwd', tar will extract it as if the name were really `etc/passwd'.

File names containing `..' can cause problems when extracting, so tar normally warns you about such files when creating an archive, and rejects attempts to extracts such files.
```
python `tarfile` might not do the same as GNU tar program, thus it has to be done manually (as we did in conan).
I may add such tests, if needed (although, I guess it's not the part of the original scope and has nothing to do with symlinks)

Reply:
@memsharded I have added two additional tests to verify security-related aspects mentioned in the https://www.gnu.org/software/tar/manual/html_node/absolute.html